src/beliefpropagation/beliefpropagation.jl

```diff
@@ -1,104 +1,82 @@
-function message_tensors(
-  tn::ITensorNetwork;
-  nvertices_per_partition=nothing,
-  npartitions=nothing,
-  subgraph_vertices=nothing,
-  kwargs...,
-)
-  return message_tensors(
-    partition(tn; nvertices_per_partition, npartitions, subgraph_vertices); kwargs...
-  )
+default_mt_constructor(inds_e) = ITensor[denseblocks(delta(inds_e))]
+default_bp_cache(ptn::PartitionedGraph) = Dict()
+function default_contractor(contract_list::Vector{ITensor})
+  return ITensor[normalize!(
+    contract(contract_list; sequence=contraction_sequence(contract_list; alg="optimal"))
+  )]
 end
 
-function message_tensors_skeleton(subgraphs::DataGraph)
-  mts = DataGraph{vertextype(subgraphs),vertex_data_type(subgraphs),ITensorNetwork}(
-    directed_graph(underlying_graph(subgraphs))
+function contract_to_MPS(contract_list::Vector{ITensor}; svd_kwargs...)
+  contract_kwargs = (;
+    alg="density_matrix",
+    output_structure=path_graph_structure,
+    contraction_sequence_alg="optimal",
+    svd_kwargs...,
   )
-  for v in vertices(mts)
-    mts[v] = subgraphs[v]
-  end
 
+  tn, _ = contract(ITensorNetwork(contract_list); contract_kwargs...)
+  mts = Vector{ITensor}(tn)
+  mts = normalize!.(mts)
   return mts
 end
 
-function message_tensors(
-  subgraphs::DataGraph;
-  itensor_constructor=inds_e -> ITensor[dense(delta(i)) for i in inds_e],
+function message_tensor(
+  ptn::PartitionedGraph, edge::PartitionEdge; mt_constructor=default_mt_constructor
 )
-  mts = message_tensors_skeleton(subgraphs)
-  for e in edges(subgraphs)
-    inds_e = commoninds(subgraphs[src(e)], subgraphs[dst(e)])
-    itensors = itensor_constructor(inds_e)
-    mts[e] = ITensorNetwork(itensors)
-    mts[reverse(e)] = dag(mts[e])
-  end
-  return mts
+  src_e_itn = subgraph(ptn, src(edge))
+  dst_e_itn = subgraph(ptn, dst(edge))
+  inds_e = commoninds(src_e_itn, dst_e_itn)
+  return mt_constructor(inds_e)
 end
 
 """
-DO a single update of a message tensor using the current subgraph and the incoming mts
+Do a single update of a message tensor using the current subgraph and the incoming mts
 """
 function update_message_tensor(
-  tn::ITensorNetwork,
-  subgraph_vertices::Vector,
-  mts::Vector{ITensorNetwork};
-  contract_kwargs=(; alg="density_matrix", output_structure=path_graph_structure, maxdim=1),
+  ptn::PartitionedGraph,
+  edge::PartitionEdge,
+  mts;
+  contractor=default_contractor,
+  mt_constructor=default_mt_constructor,
 )
-  mts_itensors = reduce(vcat, ITensor.(mts); init=ITensor[])
-
-  contract_list = ITensor[mts_itensors; ITensor[tn[v] for v in subgraph_vertices]]
-  tn = if isone(length(contract_list))
-    copy(only(contract_list))
-  else
-    ITensorNetwork(contract_list)
-  end
+  pedges = setdiff(
+    partitionedges(ptn, boundary_edges(ptn, vertices(ptn, src(edge)); dir=:in)),
```

Comment:
I think this would be good to define in a variable up above, like `partition_boundary_edges = ...` since this is a long line to read.